### ISSUE TITLE
Issue translating "Read more %s"

### ISSUE THREAD

**Issue by samikeijonen** [samikeijonen/toivo-lite#4]
This is how we're use to have `the_content` function for better accessibility. We add Post title after Continue reading or Read more. 

```
/* translators: %s: Name of current post */
the_content( sprintf(
    __( 'Continue reading %s', 'twentyfifteen' ),
    the_title( '<span class="screen-reader-text">', '</span>', false )
) );

/* translators: %s: Name of current post */
the_content( sprintf(
    __( 'Read more %s', 'toivo-lite' ),
    the_title( '<span class="screen-reader-text">', '</span>', false )
) );
```

This is the [recommended method](https://github.com/wpaccessibility/a11ythemepatterns/blob/master/read-more-links/content.php).

But @fxbenard noticed interesting issue when translating `Read more %s`. In French the translation should be like this: `Lire la suite de %s`

But it then outputs this markup:

```
<a href="#" class="more-link">Lire la suite de <span class="screen-reader-text">Post Title</span></a>
```

Instead it should be this:

```
<a href="#" class="more-link">Lire la suite <span class="screen-reader-text">de Post Title</span></a>
```

Do we have any other choices than adding html markup in the string like this:

```
/* translators: %s: Name of current post */
the_content( sprintf(
    __( 'Read more <span class="screen-reader-text">%s</span>', 'toivo-lite' ),
    the_title( '', '', false )
) );
```

**Comment by samikeijonen** [comment]
One option could be to translate it like this:
`Lire la suite <span class="screen-reader-text">de</span> %s`.

**Comment by fxbenard** [comment]
You mean

 `Lire la suite <span class="screen-reader-text">de %s</span>.`

**Comment by Otto42** [comment]
fxbenard: No, because that will create a nested span.

I guess the question is, why must the "de" be included in this span? It's not part of the post title.

**Comment by fxbenard** [comment]
Nope it's not, but "Lire la suite de" is a typo error in French. And if i don't add the "de" then it's an error for screen reader. "Lire la suite Postname", it misses the "de"

**Comment by Otto42** [comment]
But, "Lire la suite de Postname" is fine?

I just don't understand why the "de" needs to be in that span. Anything inside "screen-reader-text" is hidden. So, why should a "de" be added to the postname, specifically in this context? Should "de" always be prefixed to all uses of the post title?

I grant you that I don't understand French. :)

**Comment by fxbenard** [comment]
"Lire la suite de Postname" is fine, yes, but on front it shows : "Lire la suite de" that's not correct.

![2015-06-12_17-08-49](https://cloud.githubusercontent.com/assets/802114/8132999/c911cce0-1125-11e5-9fa2-dd2797eacbc8.jpg)

**Comment by Otto42** [comment]
Yes, I get that the "de" being shown here is bad. So, why put the "de" in at all?

The correct translation here would be "Lire la suite %s", because the "%s" is not part of the phrase "Lire la suite". It's two separate phrases in the same string. The title is on it's own off in screen-reader-text land, and it's just there for screen readers.

Should a screen-reader actually say "de Postname" or just "Postname"?

**Comment by Otto42** [comment]
Yes, I can see how that's bad. 

I would also argue that 'Read more %s' is probably a bad string in the first place. If a screen reader told me to "Read more Postname" then that's a bit weird even in English. Twenty-X uses "Continue Reading Postname" which makes somewhat more sense, but which also might have similar translation issues in some languages.

**Comment by Otto42** [comment]
I think @samikeijonen has it right here with this string:

`Lire la suite <span class="screen-reader-text">de</span> %s`

This way, the "de" is also hidden, but read as part of the whole phrase, and the nested spans are also avoided.

**Comment by fxbenard** [comment]
@Otto42 Thanks I'll go that way then. @samikeijonen your string is adopted!

**Comment by samikeijonen** [comment]
Thanks @Otto42 for valuable comments. If @fxbenard is fine with that string, I'm definitely happy:)

**Comment by joedolson** [comment]
Either adding the screen reader text class to any additional terms that should be hidden or requiring the screen reader text class in the translatable string instead of the replacement seem like valid solutions to me; the former has the advantage that it doesn't require updating the string, possibly invalidating other translations.
